### PR TITLE
PLAT-3662: Deployment IAM Policy for Terraform

### PR DESCRIPTION
**Pull request by Secretions**
Adds a policy that can be used for the actual Terraform deployment itself

deployer_image: quay.io/domino/deployer:v46

testing

https://dominodatalab.atlassian.net/browse/PLAT-3662

### CODE REVIEW

File: terraform/iam/iam-general.txt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                "arn:aws:cloudformation:<YOUR-REGION>:<IAM-ACCOUNT-NUM>:stackset/<YOUR-STACK-NAME>:*"
+            ]
+        },
+        {
+            "Sid": "S3Gated",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": [
+                "arn:aws:s3:::<YOUR-STACK-NAME>-cf-assets/*",
+                "arn:aws:s3:::<YOUR-STACK-NAME>-cf-assets"
```
Comment (ddl-jacob) [inline terraform/iam/iam-general.txt:21]: in order to tear down a deployment, these lines need to be replaced with just `"arn:aws:s3:::<YOUR-STACK-NAME>*`

File: terraform/iam/iam-general.txt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                "eks:Describe*",
+                "eks:List*",
+                "iam:Get*",
+                "iam:List*",
+                "kms:DescribeKey",
+                "s3:CreateJob",
+                "s3:Get*",
+                "s3:List*",
+                "s3:PutAccountPublicAccessBlock",
+                "s3:PutStorageLensConfiguration"
```
Comment (ddl-jacob) [inline terraform/iam/iam-general.txt:74]: need to add a line here for `"backup:ListBackupVaults"`

File: terraform/iam/iam-general.txt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            ],
+            "Resource": "*"
+        }
```
Comment (ddl-jacob) [inline terraform/iam/iam-general.txt:77]: need another section like this to clean up vaults and recovery points 
```
        {
            "Sid": "BackupGated",
            "Effect": "Allow",
            "Action": "backup:*",
            "Resource": [
                "arn:aws:backup:eu-central-1:415280201047:backup-vault:domino-dazn*",
                "arn:aws:backup:eu-central-1:415280201047:recovery-point*"
            ]
        },
```

Comment (ddl-jacob) [reply terraform/iam/iam-general.txt:77]: couldn't find any way to gate the recovery point piece to just our resources though, so might be possible to improve upon that

**Comment by Secretions** [comment]
Didn't actually merge this...

**Pull request merged**